src/js/stores/search.js

```diff
@@ -271,38 +272,46 @@ define(function (require, exports, module) {
                 var items = searchTypeInfo.getOptions();
-                
-                // Get shortest unique paths
-                var ancestors = collection.pluck(items, "pathInfo"),
-                    shortenedPaths = searchTypeInfo.shortenPaths ?
-                        pathUtil.getShortestUniquePaths(ancestors).toJS() : ancestors.toJS();
-
-                var itemMap = items.map(function (item, index) {
-                    var newPathInfo = shortenedPaths[index] || "",
-                        name = item.name,
-                        style;
-                    // Don't show the path info if it is just the same as the item name 
-                    if (name === newPathInfo) {
-                        newPathInfo = "";
-                    }
-
-                    if (name === "") {
-                        name = "Untitled";
-                        style = { "fontStyle": "italic" };
-                    }
-
-                    return {
-                        id: type + "-" + item.id.toString(),
-                        title: name,
-                        pathInfo: newPathInfo,
-                        svgType: item.iconID,
-                        category: item.category,
-                        style: style,
-                        type: "item"
-                    };
-                }.bind(this));
+                var itemMap;
+
+                if (items === null) {
+                    itemMap = [];
+                } else {
+                    // Get shortest unique paths
+                    var ancestors = collection.pluck(items, "pathInfo"),
+                        shortenedPaths = searchTypeInfo.shortenPaths ?
+                                pathUtil.getShortestUniquePaths(ancestors) : ancestors;
+
+                    itemMap = items.map(function (item, index) {
+                        var newPathInfo = shortenedPaths.get(index) || "",
+                            name = item.name,
+                            style,
+                            untitledString = nls.localize("strings.SEARCH.UNTITLED");
+                        
+                        // Don't show the path info if it is just the same as the item name 
+                        if (name === newPathInfo) {
+                            newPathInfo = "";
+                        }
+
+                        if (name === "") {
+                            name = untitledString;
+                            style = { "fontStyle": "italic" };
+                        }
+
+                        return {
+                            id: type + "-" + item.id.toString(),
+                            title: name,
+                            pathInfo: newPathInfo,
+                            svgType: item.iconID,
+                            category: item.category,
+                            style: style,
+                            type: "item"
+                        };
+                    }.bind(this));
```

Comment:
For collections functions that accept a receiver argument, that is preferred over an explicit bind.